src/Sylius/Component/Core/Provider/ProductVariantsPricesProvider.php

```diff
@@ -53,7 +54,16 @@ private function constructOptionsMap(ProductVariantInterface $variant, ChannelIn
             $optionMap[$option->getOptionCode()] = $option->getCode();
         }
 
-        $optionMap['value'] = $this->productVariantPriceCalculator->calculate($variant, ['channel' => $channel]);
+        $price = $this->productVariantPriceCalculator->calculate($variant, ['channel' => $channel]);
+        $optionMap['value'] = $price;
+
+        if ($this->productVariantPriceCalculator instanceof ProductVariantPricesCalculatorInterface) {
+            $originalPrice = $this->productVariantPriceCalculator->calculateOriginal($variant, ['channel' => $channel]);
+
+            if ($originalPrice > $price) {
+                $optionMap['original-price'] = $originalPrice;
+            }
+        }
```

Comment:
As I don't like nested `if` statements, I would go with:

```php
if (!$this->productVariantPriceCalculator instanceof ProductVariantPricesCalculatorInterface) {
    return $optionMap;
}

if ($originalPrice > $price) {
    $optionMap['original-price'] = $originalPrice;
}

return $optionMap;
```

but it's not a blocker for me 🖖

Reply:
Has @lchrusciel mentioned on the original PR, I'll try to add some original prices fixtures. I'll fix those nested `if` as well with the same PR. 

Thanks for the review !